YunoHost/yunohost
# [Fix] 1980 extract correct domain basename

## The problem

The TLD is removed from the domain basename in the line 71 by the function check_domain in the file 12-dnsrecords.pz by the following line:
`basename = domain.replace(base_dns_zone, "").rstrip(".") or "@"`

If you have a domain like “trucfrmuch.fr” the diagnosis tool will complain about not having a record for "trucmuch" if you are in France. I am located in Germany and got the same error with the domain “philippludewig” with “de” so I can expect that there are similar problems for different countries.

https://github.com/YunoHost/issues/issues/1980

## Solution

I introduced the package python3-tldextract and the following lines 
`basename = tldextract.extract(domain).domain
        if len(basename) == 0:
            basename = "@"`

## PR Status

Initialised

## How to test

Depending on your tld of your country. Example is using France.

-     *Go to the yunohost admin page to the menu domain
-     *Click on create new domain
-     Create a new domain “trucfrmuch.fr”
-     Run the diagnosis tool for it
-     See error
